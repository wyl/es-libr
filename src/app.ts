import { logger } from '@eslibr/logger'
import { AxiosProxy, ContextMiddleware } from '@eslibr/middleware'
import cors from '@koa/cors'
import Koa from 'koa'

const app = new Koa({ proxy: true })
app.use(cors())

app.use(async function (ctx, next) {
  const start = performance.now()
  await next().then(() => {
    logger.info(
      `${ctx.request.method.padEnd(10, ' ')}${ctx.req.url} ${
        ctx.response.status
      }\t${(performance.now() - start).toFixed(2)} ms`,
    )
  })
})
// import Router from "koa-router";
// export const router = new Router();

// app.use(router.routes());
// app.use(router.allowedMethods());

// !! Core Middleware Start
app.use(ContextMiddleware())
app.use(AxiosProxy())
// !! Core Middleware End

app.use((ctx) => {
  ctx.body = 'who are you?'
})

app.on('error', (err, ctx) => {
  logger.error('server error', err, ctx)
  //   logger.warn("server error", err, ctx);
  logger.info(err)
})

export { app }
