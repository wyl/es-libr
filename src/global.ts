import { INDEX_MAPPING_FILE_PATH } from "./constants";
import { IndexMappings, loadIndexMapping } from "./init/init-index-mapping";

let indexMapping: () => IndexMappings;
async function initGlobal() {
    indexMapping  = await loadIndexMapping(INDEX_MAPPING_FILE_PATH);
}

export {initGlobal, indexMapping};