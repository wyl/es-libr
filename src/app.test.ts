import request from "supertest";
import { app } from "./app";
import { initServer, stopServer } from "./global";

beforeAll(initServer);

afterAll(stopServer);

test("test", async () => {
  const response = await request(app.callback()).get("/");
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});
