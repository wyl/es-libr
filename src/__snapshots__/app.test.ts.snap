// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `"{"name":"instance-0000000000","cluster_name":"3211b412f3cc44c1a6e7eeba79a52993","cluster_uuid":"eS-GWiX4R1Kd5W1F9cG8rg","version":{"number":"8.15.3","build_flavor":"default","build_type":"docker","build_hash":"f97532e680b555c3a05e73a74c28afb666923018","build_date":"2024-10-09T22:08:00.328917561Z","build_snapshot":false,"lucene_version":"9.11.1","minimum_wire_compatibility_version":"7.17.0","minimum_index_compatibility_version":"7.0.0"},"tagline":"You Know, for Search"}"`;
